simplemma/lemmatizer.py

```diff
@@ -338,7 +74,7 @@ def _greedy_search(
     i = 0
     while candidate in datadict and (
         len(datadict[candidate]) < len(candidate)
-        and _levenshtein_dist(datadict[candidate], candidate) <= distance
+        and levenshtein_dist(datadict[candidate], candidate) <= distance
```

Comment:
How could we configure this without classes and without passing the `levenshtein_dist` method as argument to pretty much every method in this file?

Reply:
Maybe we could split this part from the rest because it's not used by non-greedy searches. The problem with the cache has been addressed and solved in #43. Feel free to open an issue if something is missing.